uaclient/security_status.py

```diff
@@ -248,30 +246,22 @@ def get_reboot_status():
     if not should_reboot():
         return RebootStatus.REBOOT_NOT_REQUIRED
 
-    try:
-        pkg_list_str = load_file(REBOOT_PKGS_FILE_PATH)
-    except FileNotFoundError:
-        # If we cannot load that file, we cannot evaluate if we have
-        # kernel related packages that require a reboot. Therefore, we
-        # will just say that a reboot is required
-        return RebootStatus.REBOOT_REQUIRED
-
-    pkg_list = pkg_list_str.split()
-    num_pkgs = len(pkg_list)
-
-    kernel_regex = "^(linux-image|linux-base).*"
-    num_kernel_pkgs = 0
+    reboot_required_pkgs = get_reboot_required_pkgs()
```

Comment:
Should we instead move all this logic into the API function and replace calls to `get_reboot_status` with calls to the API?